### PR TITLE
Dir hash mismatch 

### LINKED ISSUE #1035: no_consensus_challenge:
no_consensus_challenge: No Consensus on the challenge result. Erroring out the challenge

### PR DESCRIPTION
**Pull request by Hitenjain14**
### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- 0chain:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/validatorcore/storage/models.go [hunk 1]
```diff
@@ -52,7 +54,15 @@ func (r *DirMetaData) GetHash() string {
 	return r.Hash
 }
 
+func (r *DirMetaData) GetPath() string {
+	return r.Path
+}
+
 func (r *DirMetaData) CalculateHash() string {
+	// childPath := make([]ObjectEntity, len(r.Children))
```
Comment (lpoli) [inline code/go/0chain.net/validatorcore/storage/models.go]: Remove this comment

File: code/go/0chain.net/validatorcore/storage/models.go [hunk 2]
```diff
@@ -61,6 +71,15 @@ func (r *DirMetaData) CalculateHash() string {
 	return encryption.Hash(strings.Join(childHashes, ":"))
 }
 
+func SortChildren(children []ObjectEntity) []ObjectEntity {
```
Comment (lpoli) [inline code/go/0chain.net/validatorcore/storage/models.go]: Sorting is not required in validator.
Remove this function as well

**Review by lpoli — CHANGES_REQUESTED** [review]
@Hitenjain14 Good finding about the sort issue.
I would recommend to use collation for `path` column in `reference_objects.path` column.
Please check for appropriate collation to use for `path`.

Using `ascii(path)` cannot use index because the order is changed now. Its like defining collation at query time which makes it slow.

File: code/go/0chain.net/validatorcore/storage/models.go (hunk 1 above)
Comment (Hitenjain14) [reply code/go/0chain.net/validatorcore/storage/models.go]: Removed

File: code/go/0chain.net/validatorcore/storage/models.go (hunk 2 above)
Comment (Hitenjain14) [reply code/go/0chain.net/validatorcore/storage/models.go]: Removed

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**